hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/dynamicconfig/AddListConfigMessageTask.java

```diff
@@ -55,6 +56,9 @@ protected IdentifiedDataSerializable getConfig() {
                     (List<ItemListenerConfig>) adaptListenerConfigs(parameters.listenerConfigs);
             config.setItemListenerConfigs(itemListenerConfigs);
         }
+        MergePolicyConfig mergePolicyConfig = mergePolicyConfig(parameters.mergePolicyExist, parameters.mergePolicy,
```

Comment:
Should not we need the same check to avoid overriding what is already there when an older client connected as it done in https://github.com/hazelcast/hazelcast/pull/12299/files#diff-00fbafd5b7b8b3beea808db7f5dea4aeR49

Reply:
The default `MergePolicyConfig` returned by `AbstractAddConfigMessageTask.mergePolicyConfig()` when `parameters.mergePolicyExist == false` is the same (`new MergePolicyConfig()`) as the default of all data structure config's except for `CardinalityEstimatorConfig`, that's why the check is only present in `AddCardinalityEstimatorConfigMessageTask`.